tests/unit/macaroons/test_services.py

```diff
@@ -131,82 +106,53 @@ def test_verify_no_macaroon(self, macaroon_service):
 
     def test_verify_invalid_macaroon(self, monkeypatch, user_service, macaroon_service):
         user = UserFactory.create()
-        raw_macaroon, _ = macaroon_service.create_macaroon(
+        raw_macaroon, dm = macaroon_service.create_macaroon(
             "fake location", user.id, "fake description", {"fake": "caveats"}
         )
 
-        verifier_obj = pretend.stub(verify=pretend.call_recorder(lambda k: False))
-        verifier_cls = pretend.call_recorder(lambda *a: verifier_obj)
-        monkeypatch.setattr(services, "Verifier", verifier_cls)
+        token_obj = pretend.stub(
+            check=pretend.call_recorder(pretend.raiser(pypitoken.ValidationError)),
+            identifier=str(dm.id),
+        )
+        token_cls = pretend.stub(load=lambda *a: token_obj)
+
+        monkeypatch.setattr(pypitoken, "Token", token_cls)
 
-        context = pretend.stub()
+        context = pretend.stub(normalized_name="foo")
         principals = pretend.stub()
         permissions = pretend.stub()
 
         with pytest.raises(services.InvalidMacaroon):
             macaroon_service.verify(raw_macaroon, context, principals, permissions)
-        assert verifier_cls.calls == [
-            pretend.call(mock.ANY, context, principals, permissions)
-        ]
-
-    def test_deserialize_raw_macaroon_when_none(self, macaroon_service):
-        raw_macaroon = pretend.stub()
-        macaroon_service._extract_raw_macaroon = pretend.call_recorder(lambda a: None)
-
-        with pytest.raises(services.InvalidMacaroon):
-            macaroon_service._deserialize_raw_macaroon(raw_macaroon)
-
-        assert macaroon_service._extract_raw_macaroon.calls == [
-            pretend.call(raw_macaroon),
-        ]
-
-    @pytest.mark.parametrize(
-        "exception",
-        [
-            IndexError,
-            TypeError,
-            UnicodeDecodeError,
-            ValueError,
-            binascii.Error,
-            struct.error,
-            MacaroonDeserializationException,
-            Exception,  # https://github.com/ecordell/pymacaroons/issues/50
-        ],
-    )
-    def test_deserialize_raw_macaroon(self, monkeypatch, macaroon_service, exception):
-        raw_macaroon = pretend.stub()
-        macaroon_service._extract_raw_macaroon = pretend.call_recorder(
-            lambda a: raw_macaroon
-        )
-        monkeypatch.setattr(
-            pymacaroons.Macaroon, "deserialize", pretend.raiser(exception)
-        )
```

Comment:
`pypitoken` has a test for various types of errors https://github.com/ewjoachim/pypitoken/blob/master/tests/test_token.py#L325-L344 and a catchall because of https://github.com/ecordell/pymacaroons/issues/50 : https://github.com/ewjoachim/pypitoken/blob/master/pypitoken/token.py#L328-L332